src/misc/common_macros.hpp

```diff
@@ -0,0 +1,11 @@
+
+#ifndef QASM_SIMULATOR_COMMON_MACROS_HPP
+#define QASM_SIMULATOR_COMMON_MACROS_HPP
+
+#if defined(__GNUC__) && defined(__x86_64__)
+ #define GNUC_AVX2 1
+#else
+ #define GNUC_AVX2 0
```

Comment:
I think we can just define GNUC_AVX2 without any value, and saving the `#else` block
```suggestion
#define GNUC_AVX2
```
if `GNUC_AVX2` is not defined, then this expression will be evaluated to false.
```c++
#ifdef GNUC_AVX2
// True!
#else
// False!
#endif
```

Reply:
Absolutely!